# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/loader/CacheLoadHandler.cpp
# Commit: 3605681ebb5c
# Full Hash: 3605681ebb5cb61f1dca13c546e03367857135ae
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-11-18 15:46:32
# Regressor Bug: 1800496
# File Overlap Count: 2
# Description:
#   Bug 1800496 - Initial cancellation revert; r=asuth
#   
#   This returns most of the behavior from https://searchfox.org/mozilla-central/rev/e0a7a014f3384ac1abc229d9961f244e3e70a52c/dom/workers/ScriptLoader.cpp#735-776
#   
#   What is missing at this point is batch dispatching, as we once again cancel one-at-a-time rather than as a group.
# ==============================================================================

diff -r 6118d850a96d -r 3605681ebb5c dom/workers/loader/CacheLoadHandler.cpp
--- a/dom/workers/loader/CacheLoadHandler.cpp	Fri Nov 18 10:02:26 2022 +0000
+++ b/dom/workers/loader/CacheLoadHandler.cpp	Fri Nov 18 10:02:27 2022 +0000
@@ -280,11 +280,6 @@
 
   loadContext->mCachePromise = nullptr;
 
-  loadContext->ClearCacheCreator();
-  if (mLoader->IsCancelled() || !loadContext->mRequest) {
-    return;
-  }
-
   mLoader->LoadingFinished(mRequestHandle, aRv);
 }
 
@@ -294,11 +289,6 @@
   MOZ_ASSERT(!mRequestHandle->IsEmpty());
   WorkerLoadContext* loadContext = mRequestHandle->GetContext();
 
-  if (mLoader->IsCancelled()) {
-    Fail(NS_BINDING_ABORTED);
-    return;
-  }
-
   nsCOMPtr<nsIURI> uri;
   nsresult rv = NS_NewURI(getter_AddRefs(uri), loadContext->mRequest->mURL,
                           nullptr, mBaseURI);
@@ -360,10 +350,6 @@
   if (mFailed) {
     return;
   }
-  if (mLoader->IsCancelled()) {
-    Fail(NS_BINDING_ABORTED);
-    return;
-  }
 
   MOZ_ASSERT(loadContext->mCacheStatus == WorkerLoadContext::Uncached);
 
@@ -515,11 +501,6 @@
     return NS_OK;
   }
 
-  if (mLoader->IsCancelled() || !loadContext->mRequest) {
-    Fail(NS_BINDING_ABORTED);
-    return NS_OK;
-  }
-
   MOZ_ASSERT(loadContext->mCacheStatus == WorkerLoadContext::ReadingFromCache);
   loadContext->mCacheStatus = WorkerLoadContext::Cached;
 